### ISSUE TITLE
cli: Don't allow unknown flags

### ISSUE THREAD

**Issue by connorjclark** [GoogleChrome/lighthouse#6433]
Example, for defining multiple output types, one might mistakenly try `outputs` instead of `output`:

```
lighthouse www.example.com --outputs json html
```

This will chug along happily but only emit an html file (the default value).

**Comment by patrickhulce** [comment]
Ah this is frustrating (config*-path* gets me every time too) In general though, I'm not a huge fan of this strategy since it prevents smooth upgrades to new versions where you have a fleet of things using LH and you want your script to be able to just add the new flag and have older ones ignore it. 

Would folks find a warning just as useful or is there consensus on erroring?

**Comment by brendankenny** [comment]
I'd say at least a warning (unrecognized flag or whatever). An error might be nice, but @patrickhulce's point is a good one.

Can we take advantage of a yargs feature for this so we don't have to implement? (our version is really old, so may need to be updated)

**Comment by Hoten** [comment]
I personally want and expect cli tools to break if there's an option that is no longer supported. For renames, the old option can stick around (deprecated), and only be removed in the next major version bump.

I tested a handful of popular tools. These tools fail hard:

```
brew
npm
babel
jest
```

These tools do not:

```
yarn
```

**Comment by justinribeiro** [comment]
If an option/flag goes away, a common expectation would be to exit gracefully with either usage or help documentation. Most POSIX-compliant utilities do this per the guidelines and/or a specific coding convention (ala GNU), so it's not a huge leap.

Keeping flags around for the sake of compatibility (deprecated or not) can be troublesome over time. An example would be `sed` (which doesn't follow the more up-to-date POSIX guidelines for options and which reading the docs are almost always required).

My two cents. Take with salt.

**Comment by patrickhulce** [comment]
All true :)

I'd argue though that the situation I'm referring to is much less likely to be happening with those build tools than it is with LH. There are many situations in which people treat LH more like proto compatibility or an API call than a build tool, not to mention updates to your GNU utilities are typically *much* less frequent than updates to node modules.

Taking WPT as a prime example where on several occasions we've needed to add a flag to the LH call. If we were to implement this, then we'd need to wait for the entire worldwide fleet of WPT public + private machines to be updated to the latest version of LH (some of which are out of our control) before trying to deploy the change that takes advantage of the feature or we risk fatal errors. This feature would essentially convert every flag update into a major breaking change that needs to be manually handled.

We currently warn for unexpected audits/categories/settings, so warning here would be consistent as well.

FWIW, I'm not as concerned with keeping old flags around as I am about introducing new flags. I'm 100% onboard with a curated list of old flags that disappeared, small hamming distance spelling corrections, etc that exit with an error message.

I've made my spiel, but this certainly isn't a hill I'll die on :) exiting WFM so long as we write an apology letter to pmeenan 😆
